### PR TITLE
Dijkstra algorithm tracer class, traversal history & refinements

### PR DESCRIPTION
**Pull request by songofroland**

### CODE REVIEW

File: src/logic/dijkstra.ts [hunk 1]
```diff
@@ -1,68 +1,89 @@
...
-        if (currentCost < previousCost) {
-          costs[neighboredNode] = currentCost;
-          pathsMap[neighboredNode] = [...pathsMap[currentNode], currentNode];
+      let currentNode = this.#startNode;
+      
+      while (unvisitedNodes.size) {
+        for (const [neighbourNode, neighbourCost] of this.#graph[currentNode].entries()) {
+          if (neighbourCost !== 0) {
+            const previousCost = costs[neighbourNode];
+            const currentCost = neighbourCost + costs[currentNode];
```
Comment (wilmhit) [inline src/logic/dijkstra.ts:45]: `const currentCost = neighbourCost + previousCost;`

File: src/logic/dijkstra.ts [hunk 2]
```diff
@@ -1,68 +1,89 @@
...
+      this.paths = {};
+      this.setCostsPathsAndTraversalHistory();
+    }
 
-  while (unvisitedNodes.size) {
-    for (const [neighboredNode, neighbourCost] of graph[currentNode].entries()) {
-      if (neighbourCost !== 0) {
-        const previousCost = costs[neighboredNode];
-        const currentCost = neighbourCost + costs[currentNode];
+    setCostsPathsAndTraversalHistory() {
```
Comment (wilmhit) [inline src/logic/dijkstra.ts:32]: Am I suppose to run this from outside? Either this also gets `_` or `_getTravelsalFor` shouldn't too.

File: src/logic/dijkstra.ts [hunk 3]
```diff
@@ -1,68 +1,89 @@
...
+      this.paths = paths;
     }
-
-    unvisitedNodes.delete(currentNode);
-    currentNode = getCheapestNode(getUnvisitedNodes(costs, unvisitedNodes));
-  }
-
-  return new Result(costs, pathsMap);
+  
+    _getTraversalFor(node: number){
```
Comment (wilmhit) [inline src/logic/dijkstra.ts:63]: Return type would make it little clearer.

File: tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es6",
```
Comment (wilmhit) [inline tsconfig.json:3]: Good call

File: src/logic/dijkstra.ts [hunk 5]
```diff
@@ -1,68 +1,89 @@
...
-  const pathsMap: PathsMap = nodes.reduce(
-    (acc: PathsMap, curr: number) => (acc[curr] = [], acc), {}
-  );
-  const unvisitedNodes: Set<number> = new Set(nodes);
-  const costs: Nodes = nodes.reduce((acc: Nodes, curr: number) =>
-    (acc[curr] = Infinity, acc), {});
-
-  costs[startNode] = 0;
-  let currentNode = startNode;
+export class dijkstraTracer{
```
Comment (wilmhit) [inline src/logic/dijkstra.ts:16]: PascalCase

File: src/test/calculatedResults.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
+export const testGraph = [
```
Comment (wilmhit) [inline src/test/calculatedResults.js:1]: This file could have .ts extension looking exactly the same.

File: src/logic/dijkstra.ts (hunk 2 above)
Comment (wilmhit) [reply src/logic/dijkstra.ts:32]: _getTravelsalFor maybe actually be useful from outside in some cases.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/logic/dijkstra.ts (hunk 2 above)
Comment (songofroland) [reply src/logic/dijkstra.ts:32]: Good point, this method (`setCostsPathsAndTraversalHistory`) should be private

File: src/logic/dijkstra.ts (hunk 1 above)
Comment (songofroland) [reply src/logic/dijkstra.ts:45]: That won't work actually. Mind that the accessed attributes are different.

Comment (wilmhit) [reply src/logic/dijkstra.ts:45]: Missed that

**Review by ghost — APPROVED** [review]

**Pull request merged**